projects/kit/src/lib/processors/valid-date-preprocessor.ts

```diff
@@ -12,13 +13,25 @@ export function createValidDatePreprocessor({
     datesSeparator?: string;
 }): NonNullable<MaskitoOptions['preprocessor']> {
     return ({elementState, data}) => {
+        const {value, selection} = elementState;
+
+        if (data === dateSegmentsSeparator && value.length === selection[0]) {
```

Comment:
for which cases we need this check `value.length === selection[0]`?

Reply:
the problem is only at the end of the value, in the middle of the value the mask successfully moves the caret